draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -765,38 +764,104 @@ When defining a new `create_generators` procedure, the most important property i
+```
+
+Every `map_to_scalar` operation MUST define a unique `MAP_TO_SCALAR_ID` value to be used by the ciphersuite. This value MUST only contain ASCII encoded characters with codes between 0x21 and 0x7e (inclusive) and MUST end with an underscore (ASCII code: 0x5f), other than the last character the string MUST not contain any other underscores (ASCII code: 0x5f).
 
-### Message to Scalar as Hash
+#### Map to Scalar as Hash
 
-This operation takes an input message and maps it to a scalar value via a cryptographic hash function for the given curve. The operation takes also as an optional input a domain separation tag (dst). If a dst is not supplied, its value MUST default to the octet string returned from ciphersuite\_id || "MAP\_MSG\_TO\_SCALAR\_AS\_HASH\_", where ciphersuite\_id is the ASCII string representing the unique ID of the ciphersuite "MAP\_MSG\_TO\_SCALAR\_AS\_HASH\_" is an ASCII string comprised of 26 bytes.
+This document specifies the following `map_to_scalar` operation, called `map_to_scalar_as_hash`, that uses `hash_to_scalar` as defined in (#hash-to-scalar). Although for extendability reasons, the `map_to_scalar` operation accepts messages that can be either an octet string or a scalar value (as to support protocol specific preprocessing of a message), the `map_to_scalar_as_hash` operation used by this document only maps octet string to scalars and will abort if it gets an `input_message` that is already a scalar value. Additionally, the resulting scalar does not depend on the `index` of the message.
 
 ```
-msg_scalar = MapMessageToScalarAsHash(msg, dst)
+scalar = map_to_scalar_as_hash(msg)
 
 Inputs:
 
-- msg (REQUIRED), an octet string.
-- dst (OPTIONAL), an octet string representing a domain separation tag.
-                  If not supplied, it default to the octet string
-                  ciphersuite_id || "MAP_MSG_TO_SCALAR_AS_HASH_" where
-                  ciphersuite_id is defined by the ciphersuite.
+- msg (REQUIRED), an input_message
+
+Parameters:
+
+- dst = ciphersuite_id || "MAP_MSG_TO_SCALAR_AS_HASH_", where
+        ciphersuite_id is defined by the ciphersuite.
 
 Outputs:
 
-- msg_scalar, a scalar.
+- scalar, a scalar value.
 
 ABORT if:
 
-1. length(msg) > 2^64 - 1 or length(dst) > 255
+1. msg not an octet string
 
 Procedure:
 
 1. return hash_to_scalar(msg, dst)
 ```
 
+The `map_to_scalar` operation that will be defined by the ciphersuites of this document will be,
+
+```
+map_to_scalar(msg, index) := map_to_scalar_as_hash(msg)
+```
+
+The `MAP_TO_SCALAR_ID` of the `map_to_scalar_as_hash` operation is defines as,
+
+```
+MAP_TO_SCALAR_ID = "HM2S_"
+```
+
+### Define a new Map to Scalar
+
+To define different ways with which messages can be mapped to scalars, an application can define a new `map_to_scalar` operation, as part of a new ciphersuite. A new `map_to_scalar` function is REQUIRED to adhere to the following security rules:
+
+1. It MUST return unique values for different `msg` inputs.
```

Comment:
Within reasonable guarantees of collision avoidance?

Reply:
Updated! Thanks!